File: HackerRank 30 Days/day5/main.py
class Person:
    def __init__(self, first_name: str, last_name: str, id_number: int):
        self.first_name = first_name
        self.last_name = last_name
        self.id_number = id_number

    def printPerson(self) -> None:
        print(f"Name: {self.last_name}, {self.first_name}")
        print(f"ID: {self.id_number}")


class Student(Person):
    def __init__(self, first_name: str, last_name: str, id_number: int, scores: list[int]):
        super().__init__(first_name, last_name, id_number)
        self.scores = scores

    def calculate(self) -> str:
        average = sum(self.scores) / len(self.scores)
        if average >= 90:
            return "O"
        if average >= 80:
            return "E"
        if average >= 70:
            return "A"
        if average >= 55:
            return "P"
        if average >= 40:
            return "D"
        return "T"


def main() -> None:
    first_name, last_name, id_number = input().split()
    _ = int(input().strip())
    scores = list(map(int, input().split()))

    student = Student(first_name, last_name, int(id_number), scores)
    student.printPerson()
    print(f"Grade: {student.calculate()}")


if __name__ == "__main__":
    main()

File: HackerRank 30 Days/day5/main.cs
using System;
using System.Linq;

class Person
{
    protected string FirstName { get; }
    protected string LastName { get; }
    protected int IdNumber { get; }

    public Person(string firstName, string lastName, int idNumber)
    {
        FirstName = firstName;
        LastName = lastName;
        IdNumber = idNumber;
    }

    public void printPerson()
    {
        Console.WriteLine($"Name: {LastName}, {FirstName}");
        Console.WriteLine($"ID: {IdNumber}");
    }
}

class Student : Person
{
    private int[] Scores { get; }

    public Student(string firstName, string lastName, int idNumber, int[] scores)
        : base(firstName, lastName, idNumber)
    {
        Scores = scores;
    }

    public string calculate()
    {
        double average = Scores.Average();

        if (average >= 90) return "O";
        if (average >= 80) return "E";
        if (average >= 70) return "A";
        if (average >= 55) return "P";
        if (average >= 40) return "D";
        return "T";
    }
}

class Solution
{
    static void Main(string[] args)
    {
        string[] inputs = (Console.ReadLine() ?? string.Empty).Split(' ');
        string firstName = inputs[0];
        string lastName = inputs[1];
        int idNumber = int.Parse(inputs[2]);
        int scoreCount = int.Parse(Console.ReadLine() ?? "0");
        int[] scores = (Console.ReadLine() ?? string.Empty).Split(' ', StringSplitOptions.RemoveEmptyEntries).Select(int.Parse).ToArray();

        Student student = new Student(firstName, lastName, idNumber, scores);
        student.printPerson();
        Console.WriteLine($"Grade: {student.calculate()}");
    }
}
